security-demo/README.md

```diff
@@ -0,0 +1,263 @@
+SPIFFE recommends industry standards like TLS and JWT for forming a identity document for every service. The service-level identity AuthN and AuthZ removes the dependency of complex network-level ACL strategies.
+
+More information about SPIFFE can be found at [*https://github.com/spiffe/spiffe*](https://github.com/spiffe/spiffe).
+
+SPIRE (SPIFFE Runtime Environment) is a reference implementation of SPIFFE specification. SPIRE manages identities for node and workloads.
+ It provides API for controlling attestation policies and identity issuance and rotation strategies.
+
+More information about SPIRE can be found at [*https://github.com/spiffe/spire*](https://github.com/spiffe/spire).
+
+## Benefits
+
+* Node attestation: Only verifiable edge nodes can join the edge clusters. Every node is issued an identity on verification.
+ In case of failed node attestations, no identity documents can be issued for services running on the node.
+
+* Workload attestation: Only verifiable workload can run on edge nodes. In case of failed workload attestations, there are no identities issues for the workloads.
+All communications are blocked from unverified workloads.
+
+* Certificate rotation: Short-lived certificates are generated and rotation policies can be configured for every service communication.
+There is no need for custom agents and reliance on specific orchestrators for certificate rotation configuration and management.
+
+* Automated non-root CA certificate heirarchical deployments: Edge spire servers can be configured to not share any root CA chain for downstream nodes and workloads.
+
+## SPIRE-based Identity Management Reference Architecture
+
+Goal of this example is to achieve workload identity management for
+cloud and edge processes . A reference architecture is shown below. A
+list of things to-do is listed below in version information section.
+For further integration , an extension add-on will be developed
+integrated with kubeege for the same.
+
+![image](./doc/images/demo-arch.png)
+
+## Source Folders
+* release : Directory has configurations and scripts to be used for deployment of identity management infrastructure for Kubeedge.
+
+* app-agent-conf  : Configurations for spire agent interfacing with edge (event-bus) and user applications . Spire agent communicates to edge spire server.|
+
+* conf : Configurations for spire server (cloud and edge) and spire agent (agent communicating to cloud spire server).
+
+* certs : Spiffe helper communicates with spire agents and download certificates in this folder . Ghostunnel is loaded with these certificates . For edge to user interface application , spiffe helper configuration and ghostunnel certs are present in event-bus and user-app folders respectively.
+
+* app-binaries : All binaries for edge , cloud and test applications.
```

Comment:
I am wondering if this folder is required?

Reply:
The folder will contain binaries when release is built. No binaries are uploaded directly.